Failure to compile on Power due to __bool

I had someone report that my code using stdexec didn't compile on a Power9 system.

```
__detail/__meta.hpp:71:7: error: expected identifier before '__attribute__'
    using __bool = std::bool_constant<_B>;
```

At first this looked a little confusing since there is no `__attribute__` here, but I think what is going on is that `__bool` is defined on Power as a compiler intrinsic (if that is the right word here).  See https://gcc.gnu.org/onlinedocs/gcc/PowerPC-AltiVec_002fVSX-Built-in-Functions.html :

> Compiling with -maltivec adds keywords __vector, vector, __pixel, pixel, __bool and bool. When compiling ISO C, the context-sensitive substitution of the keywords vector, pixel and bool is disabled. To use them, you must include <altivec.h> instead.

Should I send up a PR that changes all `__bool` to `__boolean` ?  Is there something else that should be used instead of `stdexec::__bool`.

--- a/include/exec/static_thread_pool.hpp
+++ b/include/exec/static_thread_pool.hpp
@@ -110,7 +110,7 @@ namespace exec {
       template <class Fun, class Shape, class... Args>
         requires stdexec::__callable<Fun, Shape, Args...>
       using bulk_non_throwing = //
-        stdexec::__bool<
+        stdexec::__mbool<
           // If function invocation doesn't throw
           stdexec::__nothrow_callable<Fun, Shape, Args...> &&
           // and emplacing a tuple doesn't throw

--- a/include/exec/when_any.hpp
+++ b/include/exec/when_any.hpp
@@ -40,7 +40,7 @@ namespace exec {
     using __signature_to_tuple_t = decltype(__signature_to_tuple_((_Sig*) nullptr));
 
     template <class... _Args>
-    using __all_nothrow_decay_copyable = __bool<(__nothrow_decay_copyable<_Args> && ...)>;
+    using __all_nothrow_decay_copyable = __mbool<(__nothrow_decay_copyable<_Args> && ...)>;
 
     template <class... _Args>
     using __all_nothrow_move_constructible =

--- a/include/nvexec/stream/when_all.cuh
+++ b/include/nvexec/stream/when_all.cuh
@@ -123,7 +123,7 @@ namespace nvexec::STDEXEC_STREAM_DETAIL_NS {
 
       template <class Completions>
       using sends_values = //
-        stdexec::__bool<
+        stdexec::__mbool<
           stdexec::__v< stdexec::__gather_signal<
             stdexec::set_value_t,
             Completions,

--- a/include/stdexec/__detail/__meta.hpp
+++ b/include/stdexec/__detail/__meta.hpp
@@ -70,7 +70,7 @@ namespace stdexec {
   using __t = typename _Tp::__t;
 
   template <bool _Bp>
-  using __bool = std::bool_constant<_Bp>;
+  using __mbool = std::bool_constant<_Bp>;
 
   template <class _Ty>
   struct __mtype {
@@ -344,7 +344,7 @@ namespace stdexec {
   template <class _Tp>
   struct __contains {
     template <class... _Args>
-    using __f = __bool<(__v<std::is_same<_Tp, _Args>> || ...)>;
+    using __f = __mbool<(__v<std::is_same<_Tp, _Args>> || ...)>;
   };
 
   template <class _Continuation = __q<__types>>
@@ -564,7 +564,7 @@ namespace stdexec {
   };
   template <bool _Enable, class _Fun, class... _As>
   using __call_result_if_t =
-    typename __if<__bool<_Enable>, __qcall_result, __>::template __f<_Fun, _As...>;
+    typename __if<__mbool<_Enable>, __qcall_result, __>::template __f<_Fun, _As...>;
 
   // For emplacing non-movable types into optionals:
   template <class _Fn>
@@ -660,11 +660,11 @@ namespace stdexec {
   };
 
   template <class... _Booleans>
-  using __mand = __bool<(__v<_Booleans> && ...)>;
+  using __mand = __mbool<(__v<_Booleans> && ...)>;
   template <class... _Booleans>
-  using __mor = __bool<(__v<_Booleans> || ...)>;
+  using __mor = __mbool<(__v<_Booleans> || ...)>;
   template <class _Boolean>
-  using __mnot = __bool<!__v<_Boolean>>;
+  using __mnot = __mbool<!__v<_Boolean>>;
 
   template <class _Fn>
   struct __mall_of {
@@ -719,7 +719,7 @@ namespace stdexec {
   using __2 = __placeholder<2>;
   using __3 = __placeholder<3>;
 
-  template <class _Ty, class _Noexcept = __bool<true>>
+  template <class _Ty, class _Noexcept = __mbool<true>>
   struct __mconstruct {
     template <class... _As>
     auto operator()(_As&&... __as) const noexcept(__v<_Noexcept>&& noexcept(_Ty((_As&&) __as...)))
@@ -728,7 +728,7 @@ namespace stdexec {
     }
   };
 
-  template <template <class...> class _Cp, class _Noexcept = __bool<true>>
+  template <template <class...> class _Cp, class _Noexcept = __mbool<true>>
   using __mconstructor_for = __mcompose<__q<__mconstruct>, __q<_Cp>>;
 
   template <std::size_t>
@@ -822,7 +822,7 @@ namespace stdexec {
   using __dispatch_result_t = __call_result_t<__mdispatch<_Ty>, _Ts...>;
 
   template <class _Signature, class... _Args>
-  using __try_dispatch_ = __bool<__dispatchable<_Signature, _Args...>>;
+  using __try_dispatch_ = __mbool<__dispatchable<_Signature, _Args...>>;
 
   template <class _Signatures, class _Continuation = __q<__mfront>>
   struct __which { };

--- a/include/stdexec/execution.hpp
+++ b/include/stdexec/execution.hpp
@@ -579,7 +579,7 @@ namespace stdexec {
     template <class _Receiver, class _Tag, class... _Args>
     using __missing_completion_signal_t = //
       __if<
-        __bool<nothrow_tag_invocable<_Tag, _Receiver, _Args...>>,
+        __mbool<nothrow_tag_invocable<_Tag, _Receiver, _Args...>>,
         __found_completion_signature,
         _MISSING_COMPLETION_SIGNAL_<_Tag(_Args...)>>;
 
@@ -868,7 +868,7 @@ namespace stdexec {
     sender_in<_Sender, _Env> && __valid<__single_value_variant_sender_t, _Sender, _Env>;
 
   template <class... Errs>
-  using __nofail = __bool<sizeof...(Errs) == 0>;
+  using __nofail = __mbool<sizeof...(Errs) == 0>;
 
   template <class _Sender, class _Env = no_env>
   concept __nofail_sender =
@@ -1048,7 +1048,7 @@ namespace stdexec {
   template <class _Tag, const auto& _Predicate>
   concept tag_category = //
     requires {
-      typename __bool<bool{_Predicate(_Tag{})}>;
+      typename __mbool<bool{_Predicate(_Tag{})}>;
       requires bool{_Predicate(_Tag{})};
     };
 
@@ -2699,7 +2699,7 @@ namespace stdexec {
 
   template <class _Fun, class... _Args>
     requires invocable<_Fun, _Args...>
-  using __non_throwing_ = __bool<__nothrow_invocable<_Fun, _Args...>>;
+  using __non_throwing_ = __mbool<__nothrow_invocable<_Fun, _Args...>>;
 
   template <class _Tag, class _Fun, class _Sender, class _Env>
   using __with_error_invoke_t = //
@@ -4775,7 +4775,7 @@ namespace stdexec {
         }
 
         template <class... _Errs>
-        using __all_nothrow_decay_copyable = __bool<(__nothrow_decay_copyable<_Errs> && ...)>;
+        using __all_nothrow_decay_copyable = __mbool<(__nothrow_decay_copyable<_Errs> && ...)>;
 
         template <class _Env>
         using __scheduler_with_error_t = //
@@ -5289,7 +5289,7 @@ namespace stdexec {
         __minvoke< __mconcat<__qf<set_value_t>>, __single_values_of_t<_Env, _Senders>...>>;
 
     template <class... _Args>
-    using __all_nothrow_decay_copyable = __bool<(__nothrow_decay_copyable<_Args> && ...)>;
+    using __all_nothrow_decay_copyable = __mbool<(__nothrow_decay_copyable<_Args> && ...)>;
 
     template <class _Env, class... _SenderIds>
     using __all_value_and_error_args_nothrow_decay_copyable = __mand<

--- a/include/tbbexec/tbb_thread_pool.hpp
+++ b/include/tbbexec/tbb_thread_pool.hpp
@@ -89,7 +89,7 @@ namespace tbbexec {
         };
 
         template <class Fun, class Shape, class... Args>
-        using bulk_non_throwing = stdexec::__bool<
+        using bulk_non_throwing = stdexec::__mbool<
           // If function invocation doesn't throw
           stdexec::__nothrow_callable<Fun, Shape, Args...> &&
           // and emplacing a tuple doesn't throw

